src/core/Akka/IO/TlsConnection.cs

```diff
@@ -0,0 +1,1183 @@
+﻿// //-----------------------------------------------------------------------
+// // <copyright file="TlsConnection.cs" company="Akka.NET Project">
+// //     Copyright (C) 2009-2020 Lightbend Inc. <http://www.lightbend.com>
+// //     Copyright (C) 2013-2020 .NET Foundation <https://github.com/akkadotnet/akka.net>
+// // </copyright>
+// //-----------------------------------------------------------------------
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Net.Security;
+using System.Net.Sockets;
+using System.Runtime.CompilerServices;
+using System.Security.Cryptography.X509Certificates;
+using System.Threading.Tasks;
+using Akka.Actor;
+using Akka.Dispatch;
+using Akka.Event;
+using Akka.IO.Buffers;
+using Akka.Pattern;
+using Akka.Util;
+
+namespace Akka.IO
+{
+    internal class SendReceiveArgsFake
```

Comment:
SslStream doesn't have `SocketAsyncEventArgs`, so to keep this class as similar as possible to Tcp we use this `SendReceiveArgsFake` to mimic the abstraction